streams/src/main/java/org/apache/kafka/streams/processor/internals/TaskManager.java

```diff
@@ -454,6 +479,24 @@ private void handleTasksWithoutStateUpdater(final Map<TaskId, Set<TopicPartition
                                                 final Map<TaskId, Set<TopicPartition>> standbyTasksToCreate,
                                                 final Map<Task, Set<TopicPartition>> tasksToRecycle,
                                                 final Set<Task> tasksToCloseClean) {
+        final Map<Task, Set<TopicPartition>> initialStandbyTasksToRecycle = assignInitialTasks(activeTasksToCreate, logPrefix, topologyMetadata, changelogReader);
+        final Map<Task, Set<TopicPartition>> initialStandbyTasksToUse = assignInitialTasks(activeTasksToCreate, logPrefix, topologyMetadata, changelogReader);
```

Comment:
Should we pass in `standbyTasksToCreate` instead of `activeTasksToCreate`? (If yes, seems we would need better testing to cover this case?)

Reply:
Good spot. I'd already fixed this in the `StateUpdater` implementation (`handleInitialTaskReuse`), which was picked up by a test, but this one must have slipped through.

You're right that this needs better test coverage, so I'll leave this comment open as a reminder to add a test here.

(I'm working on fixing this right now, but having some problems with failing tests)

Reply:
While testing this I saw some sporadic test failures of `StreamThreadTest`, due to what looked like a race condition/deadlock in TaskExecutor shutdown. However, I could only reproduce it very rarely (> 1 in 20 runs), so it's likely unrelated to this fix and instead a bug in `trunk`. Will keep an eye on it though.

Reply:
I've added some tests around Task management here: https://github.com/apache/kafka/pull/16922/commits/22a1284e6965d7a38f4e76e00c5191acc2e53734
I've verified that these tests fail in the presence of this typo, and also fail if I remove the implementation that assigns pending standbys.